modules/engage-paella-player/src/main/paella-opencast/plugins/es.upv.paella.opencast.loader/01_prerequisites.js

```diff
@@ -79,20 +79,30 @@ class Opencast {
     return this.getEpisode()
     .then(function(episode) {
       return new Promise((resolve, reject)=>{
-        var serie = episode.mediapackage.series;
-        if (serie != undefined) {
-          base.ajax.get({url:'/series/'+serie+'.json'},
-            function(data,contentType,code) {
-              self._series = data;
-              resolve(self._series);
-            },
-            function(data, contentType, code) {
-              reject();
-            }
-          );
-        }
-        else {
-          reject();
+        if (self._series) {
+	  resolve(self.series);
+	}
+	else {
+          var serie = episode.mediapackage.series;
+          if (serie != undefined) {
+            base.ajax.get({url:'/search/series.json', params:{'id': serie}},
+              function(data, contentType, code) {
+                if (data['search-results'].result) {
+                  self._series = data['search-results'].result;
```

Comment:
Note that result can either be an array or a single element. You probably want to ensure it's always an array.
E.g. something along the lines of:

```
data = data['search-results'].result;
self._series = Array.isArray(data.result) ? data.result : [data.result];
```

Reply:
Generally makes sense. I think there might be a problem though if you only have one series.

Also, could you please add a proper description to this pull request (e.g. reasoning for the change)? Copying the text over from Jira should be fine but it's really annoying as a reviewer to try finding information about pull requests in other sources. Be it Jira, be it the mailing list, … ;-P

Reply:
As the call is specifying the id of the series (?id=<serieid>) the call returns always a single element, so there is no need to convert it to an array. The function returns the series element of the episode.

Reply:
Hi @lkiesow , I can add my 2¢ on the benefit of avoiding the series service from the Player. Traditionally, for distributed Opencast systems that do not have an independent Solr server, the Series Impl is on the Admin node and the Series Remote is on the Engage node. When a call comes in from the Player to the series remote endpoint on the Engage node, the remote service uses the ServiceRegistry to locate a series Impl service that has availability to handle a series service inquiry.

Two issues
1. When admin is down the series remote service is out of service. This makes the player dependent on both the Engage and the Admin node.
2. The following ServiceRegistry method does a search for *all* services and the current load of *all* services and the current availability of *all* services running the Opencast system. This method takes multiple seconds to gather data, and even longer in a busy system. Once the series Impl is finally parsed out of all the gathered service data, the actual call from the remote to the impl takes only milliseconds. 

Not sure how to solve the expense of the getServiceRegistrationsByLoad() method. In the meantime, it's simplier and easlier for new implementers if the Player quickly retrieves locally accessible series metadata from the search endpoint. The search endpoint has all required series metadata from any mediapackage that has been previously published with a series association.

https://github.com/opencast/opencast/blob/ceb51e5a42d21997f478b0d84ef674ab4fc310ee/modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/ServiceRegistryJpaImpl.java#L2166-L2187

Reply:
> As the call is specifying the id of the series (?id=) the call returns always a single element, 

makes sense